homeassistant/components/airvisual/config_flow.py

```diff
@@ -69,6 +69,18 @@ async def async_step_user(self, user_input=None):
 
         geo_id = async_get_geography_id(user_input)
         await self._async_set_unique_id(geo_id)
+        self._abort_if_unique_id_configured()
+
+        # Find older config entries without unique ID
+        for entry in self._async_current_entries():
```

Comment:
@bachya  With the extra waiting, the test discovered an actual bug in the airvisual config flow, which I fixed here.